test/e2e/security_test.go

```diff
@@ -169,29 +167,37 @@ func testCert(t *testing.T, data *TestData, expectedCABundle string, restartPod
 			t.Log("CA bundle doesn't match the expected one, retrying")
 			return false, nil
 		}
-		clientConfig := restclient.Config{
-			TLSClientConfig: restclient.TLSClientConfig{
-				Insecure:   false,
-				ServerName: certificate.GetAntreaServerNames(certificate.AntreaServiceName)[0],
-				CAData:     []byte(caBundle),
+
+		// We create a test Pod which needs access to the CA bundle.
+		// Because this Pod will be created inside the test Namespace, it will not have direct access to the antrea-ca ConfigMap,
+		// which is in the kube-system Namespace. Therefore, we make a "copy" of the antrea-ca ConfigMap in the test Namespace.
+		cm := corev1.ConfigMap{
```

Comment:
I just realized that ConfigMap creation and test Pod creation happens inside of one big `wait.Poll`. I don't think this is necessary or even a good idea.
IMO, we should have something like this:
1) `wait.Poll` to check for CA bundle availability
2) create ConfigMap copy
3) create test Pod
4) `wait.Poll` for curl command

cc @luolanzone

Reply:
@KMAnju-2021 I didn't see any update for a long time, could you address this comment so we can move forward? thanks.